core/node/storage/test_utils.go

```diff
@@ -0,0 +1,87 @@
+package storage
+
+import (
+	"io/fs"
+)
+
+type ReadDirFileFS interface {
+	fs.FS
+	fs.ReadDirFS
+	fs.ReadFileFS
+}
+
+// LayeredFS represents a filesystem that combines two embed.FS instances,
+// with the primary taking precedence over the fallback
+type LayeredFS struct {
+	primary  ReadDirFileFS
+	fallback ReadDirFileFS
+}
+
+var _ ReadDirFileFS = (*LayeredFS)(nil)
+
+// New creates a new LayeredFS with the given primary and fallback filesystems
+func NewLayeredFS(primary, fallback ReadDirFileFS) *LayeredFS {
+	return &LayeredFS{
+		primary:  primary,
+		fallback: fallback,
+	}
+}
+
+// Open implements fs.FS interface
+func (l *LayeredFS) Open(name string) (fs.File, error) {
+	// Try to open from primary first
+	if file, err := l.primary.Open(name); err == nil {
```

Comment:
If it is used in test code only this is probably fine. But you should check that an error when dealing with a file on the primary is the result of a file not found before considering the fallback fs.